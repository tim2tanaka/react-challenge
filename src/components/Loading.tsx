import { useAppState } from '../app_state/context';
import spinner from '../spinner.svg';

export function Loading() {
  const state = useAppState();
  const loadingClass = state.isloading ? 'loading' : 'loading-hide';
  return (
    <div className={loadingClass}>
      <img src={spinner} className="loading-logo" alt="loading-logo" />
    </div>
  );
}
