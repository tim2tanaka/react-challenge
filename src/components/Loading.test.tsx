import { render } from '@testing-library/react';
import { Loading } from './Loading';

describe('Loading', () => {
  test('render Loading', () => {
    const { container } = render(<Loading />);
    expect(container).toMatchSnapshot();
  });
});
