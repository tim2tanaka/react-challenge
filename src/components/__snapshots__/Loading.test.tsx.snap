// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading render Loading 1`] = `
<div>
  <div
    class="loading-hide"
  >
    <img
      alt="loading-logo"
      class="loading-logo"
      src="spinner.svg"
    />
  </div>
</div>
`;
